### PR TITLE
fix: Prevent Creating Groups With No Name

### LINKED ISSUE #2800: [BUG] - Lack of Group Naming Validation
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [X] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

Mealie allows users to create groups without assigning them a name. 

Once created it is not possible to move users into the group, because for user creation or moving into a new group the group name is required, but an empty string will not be accepted. 

### Steps to Reproduce

Via the Gui: 
- visit `/admin/manage/groups` page
- Click **Create** Button
- Click Create without doing anything

Via the API
- post a request to `/api/admin/groups` with the content `{ "name": "" }`
- get this response:
```
{
  "name": "",
  "id": "8944b068-71e6-40ed-9855-037621d76fd0",
  "slug": "",
  "categories": [],
  "webhooks": [],
  "users": [],
  "preferences": null
}
```

### Please provide relevant logs

Just a 201

[12/08/23 09:30:39] INFO     127.0.0.1:0 - "POST /api/admin/groups HTTP/1.1" 201 

### Mealie Version

Issue can be reproduced in my dev container and the demo

### Deployment

Docker (Linux), Docker (Windows)

### Additional Deployment Details

_No response_

### PR DESCRIPTION
**Pull request by michael-genson**
## What type of PR is this?

_(REQUIRED)_

- bug

## What this PR does / why we need it:

_(REQUIRED)_

This makes it so groups can't have an empty name (i.e. `name == ""`). Prior to this PR an empty group name is disallowed in some places, but not at group creation time.

I also added a db fix to check if a group with no name already exists (and rename it if it does).

## Which issue(s) this PR fixes:

_(REQUIRED)_

Fixes #2800

## Special notes for your reviewer:

Horses with no name are still allowed.

## Testing

_(fill-in or delete this section)_

I tested the db fix manually (disabled the validator, created a group with no name, restarted the db, observed the fix). I'm not sure why the frontend allows you to submit the form with an empty name (it's set as required) but at least now it won't work (which I also tested, just to be sure).

### CODE REVIEW

File: mealie/db/fixes/fix_group_with_no_name.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+            counter = i + offset
+            try:
+                _do_fix(session, group, counter)
+                break
+            except IntegrityError:
+                session.rollback()
+                attempts += 1
+                offset += 1
+                continue
```
Comment (hay-kot) [inline mealie/db/fixes/fix_group_with_no_name.py:50]: Instead of a integer offset, could we just use part of a UUID? That way it's sudo random and you have less chances of collision. 

Or we just use a UUID for the name. Either way I'm good with

File: mealie/schema/user/user.py [hunk 2]
```diff
@@ -66,6 +66,14 @@ class GroupBase(MealieModel):
...
         orm_mode = True
 
+    @validator("name")
+    def not_null(cls, v: str):
+        v = v.strip()
+        if not v:
+            raise ValueError("name cannot be empty")
+
+        return v
+
```
Comment (hay-kot) [inline mealie/schema/user/user.py]: Probably easier to do this with a constrained type

https://docs.pydantic.dev/1.10/usage/types/#constrained-types

tldr:

```python
class Model(BaseModel):
    upper_str: constr(to_upper=True)
    lower_str: constr(to_lower=True)
    short_str: constr(min_length=2, max_length=10)
    regex_str: constr(regex=r'^apple (pie|tart|sandwich)$')
    strip_str: constr(strip_whitespace=True)
```

**Review by hay-kot — CHANGES_REQUESTED** [review]

File: mealie/db/fixes/fix_group_with_no_name.py (hunk 1 above)
Comment (michael-genson) [reply mealie/db/fixes/fix_group_with_no_name.py:50]: Oh duh, yeah I'll just use the id as the name, thanks!

File: mealie/schema/user/user.py (hunk 2 above)
Comment (michael-genson) [reply mealie/schema/user/user.py]: Huh, that's a nice feature, TIL

**Review by hay-kot — APPROVED** [review]
